### ISSUE TITLE
Warnings generated by Post Checkout Code step

### ISSUE THREAD

**Issue by DanRStevens** [actions/checkout#805]
On the Summary page for my workflow I see the following Annotations at the bottom:
> The process '/usr/bin/git' failed with exit code 128

![image](https://user-images.githubusercontent.com/16793474/170150845-20c85139-4ebc-414f-81ce-e5835852cdaf.png)

Following the link to the `jobs` details shows an error in the **Post Checkout Code** step at the end of each job. (I edited "projectName" below):
> Post job cleanup.
> /usr/bin/git version
> git version 2.36.1
> Temporarily overriding HOME='/home/runner/work/_temp/24a23bdc-b10a-4c41-843c-740993d4f320' before making global git config changes
> Adding repository directory to the temporary git global config as a safe directory
> /usr/bin/git config --global --add safe.directory /home/runner/work/_projectName_/_projectName_
> /usr/bin/git config --local --name-only --get-regexp core\.sshCommand
> /usr/bin/git submodule foreach --recursive git config --local --name-only --get-regexp 'core\.sshCommand' && git config --local --unset-all 'core.sshCommand' || :
> fatal: No url found for submodule path '_projectName_' in .gitmodules
> Warning: The process '/usr/bin/git' failed with exit code 128

We are not using any git submodules in this repository. The name listed is the name of the repo holding all the project code.

No idea why the warning is being generated. The workflow appears to run fine. By that point in the workflow all of the code I wrote will have already completed. I'm not even sure why there needs to be a "Post Checkout Code" step.

**Comment by DanRStevens** [comment]
Just in case this is relevant, there was some code added in the last week or so where one of the main build jobs tries to re-use another workflow form the same repository.
```yml
backup:
  ...
  uses: ./.github/workflows/backup.yml
  ...
```

I don't remember seeing the warning before, so maybe the warning is new since that code addition.

**Comment by jsoref** [comment]
This is technically a duplicate of #354

**Comment by DanRStevens** [comment]
I see that related issue has been closed, though people are still complaining it's an issue for them.

In the related issue, most comments are about self hosted runners. I'm using GitHub hosted runners.

Issue still seems relevant. I still see warnings on the GitHub Actions run summary page.

**Comment by jsoref** [comment]
I don't suppose you could provide a sample workflow/repository/log?

**Comment by DanRStevens** [comment]
I've prepared a minimal workflow that is able to reproduce the warning:
```yaml
name: Checkout

on:
  push:

jobs:
  checkout:
    runs-on: ubuntu-22.04

    steps:
      - name: Checkout Code
        uses: actions/checkout@v3
```

In the run logs, the warning appears in the "Post Checkout Code" section ("projectName" was edited):
> Post job cleanup.
/usr/bin/git version
git version 2.38.2
Temporarily overriding HOME='/home/runner/work/_temp/a391c45c-1a83-4d06-b293-db9a30a8387e' before making global git config changes
Adding repository directory to the temporary git global config as a safe directory
/usr/bin/git config --global --add safe.directory /home/runner/work/projectName/projectName
/usr/bin/git config --local --name-only --get-regexp core\.sshCommand
/usr/bin/git submodule foreach --recursive sh -c "git config --local --name-only --get-regexp 'core\.sshCommand' && git config --local --unset-all 'core.sshCommand' || :"
fatal: No url found for submodule path 'projectName' in .gitmodules
Warning: The process '/usr/bin/git' failed with exit code 128

On the run Summary page, an Annotation is seen with the warning:
> Annotations
1 warning
checkout
The process '/usr/bin/git' failed with exit code 128

![image](https://user-images.githubusercontent.com/16793474/211461005-d0d77226-b18b-4e7b-9720-1cd286878d56.png)

**Comment by DanRStevens** [comment]
So we found the cause of the error, and it looks to have been on our end.

I thought we weren't using any submodules, since we don't generally use them where I'm working now, and there was no `.gitmodules` file. However, it appears someone had accidentally added a submodule, and committed the empty folder, but not the corresponding `.gitmodules` file. This is what caused the error. Curiously, it only caused an error in the "Post Checkout Code" step.

Steps to reproduce:
1. Create an empty repo.
2. Add the minimal workflow posted above
3. Push change, verify there is no warning
4. Commit a half created submodule reference:
    - `git submodule add ./ submodule`
    - `git reset .gitmodules`
    - Commit
5. Push change, verify the warning appeared
6. Commit the `.gitmodules` file
7. Push change, verify the warning is now gone

So in summary, the problem was on our end. Sorry to bother you.
